service/src/integrationTest/kotlin/fi/espoo/evaka/ChildBuilder.kt

```diff
@@ -0,0 +1,280 @@
+package fi.espoo.evaka
+
+import fi.espoo.evaka.application.ApplicationStatus
+import fi.espoo.evaka.daycare.service.AbsenceType
+import fi.espoo.evaka.daycare.service.CareType
+import fi.espoo.evaka.placement.PlacementType
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.dev.DevAssistanceNeed
+import fi.espoo.evaka.shared.dev.DevChild
+import fi.espoo.evaka.shared.dev.DevPerson
+import fi.espoo.evaka.shared.dev.insertTestAbsence
+import fi.espoo.evaka.shared.dev.insertTestApplication
+import fi.espoo.evaka.shared.dev.insertTestAssistanceNeed
+import fi.espoo.evaka.shared.dev.insertTestBackUpCare
+import fi.espoo.evaka.shared.dev.insertTestChild
+import fi.espoo.evaka.shared.dev.insertTestDaycareGroupPlacement
+import fi.espoo.evaka.shared.dev.insertTestNewServiceNeed
+import fi.espoo.evaka.shared.dev.insertTestPerson
+import fi.espoo.evaka.shared.dev.insertTestPlacement
+import fi.espoo.evaka.shared.dev.insertTestPlacementPlan
+import fi.espoo.evaka.shared.domain.FiniteDateRange
+import java.lang.IllegalStateException
+import java.time.LocalDate
+import java.util.UUID
+
+class ChildBuilder(
+    private val tx: Database.Transaction,
+    private val today: LocalDate
+) {
+    fun childOfAge(years: Int, months: Int = 0, days: Int = 0): ChildFixture {
+        val dob = today.minusYears(years.toLong()).minusMonths(months.toLong()).minusDays(days.toLong())
+        val childId = tx.insertTestPerson(DevPerson(dateOfBirth = dob))
+        tx.insertTestChild(DevChild(childId))
+        return ChildFixture(tx, today, childId)
+    }
+
+    class ChildFixture(
+        private val tx: Database.Transaction,
+        private val today: LocalDate,
+        val childId: UUID
+    ) {
+        fun hasAssistanceNeed() = AssistanceNeedBuilder(tx, today, this)
```

Comment:
Maybe some other name would be better? Like `withAssistanceNeed` or `addAssistanceNeed`.

Reply:
Yeah `add` could work. I reserved `with` for hierarchical chaining, e.g. `add` placement `with` service need.

Reply:
Well, there _is_ a hierarchy between children and their placements 😄

Reply:
Hah well thats true.. But if it was 
```
child
   .withAbsence() ...
   .withPlacement() ...
      .withServiceNeed() ...
```
then without indentation it would not be clear that service need is child of placement but placement is not child of absence.

Hmm how about this?
```
child
   .addAbsence() ... .save()
   .addPlacement() ... .save()
   .addPlacement() ... .save { placement ->
      placement.addServiceNeed() ... save()
      placement.addGroupPlacement() ... save()
   }
   .addAssistanceNeed() ... .save()
```
Then for the sake of consistency the child save maybe should also take a block function, but maybe it's ok to handle root level differently.

Reply:
I think that's a good idea. Another possible solution would be to make `addServiceNeed` a part of `PlacementBuilder` and "saving" the service need would just store a callback that persists the service need in `PlacementBuilder` that would be called after the placement is persisted. So things would be written to the database only when the whole "building process" is done.